pyproject.toml

```diff
@@ -1,68 +1,39 @@
-[tool.black]
-line-length = 120
-
-[tool.pycln]
-all = true
-
-[tool.pylint."messages control"]
-disable = ["all"]
-enable = [
-    "abstract-method",
-    "bad-staticmethod-argument",
-    "consider-merging-isinstance",
-    "function-redefined",
-    "undefined-variable",
-    "unused-private-member",
-]
-
-[tool.ruff]
-# Configure the line-length so Ruff knows how long to make any lines it
-# modifies.
-line-length = 120
-target-version = "py38"
-
-select = ["F", "E", "EM"]
-
-ignore = [
-
-  "E501", # E501 line length (we use black auto-formatter to handle line length)
-  "E741", #	ambiguous-variable-name
-]
-
-[tool.ruff.pydocstyle]
-convention = "google"
-
-[tool.pydoclint]
-style = 'google'
-exclude = '\.git|\.tox|tests/.'
-allow-init-docstring = true
-
 [build-system]
-requires = ["hatchling", "hatch-requirements-txt"]
+requires = [
+    "hatchling",
+    "hatch-requirements-txt"
+]
 build-backend = "hatchling.build"
 
 [project]
 name = "tecton-client"
+dependencies = ["httpx"]
```

Comment:
idea: we could set a min httpx version just to have a reasonable floor for what we support `0.23.0` was released May 2022. That seems like a fair "floor" version for us to support?